cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/ApacheSparkContext.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.spark;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.apache.spark.SparkContext;
+import org.apache.spark.api.java.JavaSparkContext;
+
+import java.util.Map;
+
+/**
+ * Interface to provide abstraction for accessing either scala or java version of the Apache SparkContext.
+ */
+interface ApacheSparkContext {
```

Comment:
Name is a little misleading, because this has a method getContext() that returns an Apache SparkContext - according to its javadoc. Can we call this a SparkFacade or if you have a better idea...